client/webserver/webserver.go

```diff
@@ -136,26 +138,27 @@ func New(core clientCore, addr string, logger dex.Logger, reloadHTML bool) (*Web
 	execPath = filepath.Dir(execPath) // e.g. /opt/decred/dex
 
 	// executable path
-	siteDir := filepath.Join(execPath, "site") // e.g. /opt/decred/dex/site
-	log.Debugf("Looking for site in %v", siteDir)
-	if !folderExists(siteDir) {
-
-		// working directory
-		siteDir, _ = filepath.Abs("site")
-		log.Debugf("Looking for site in %v", siteDir)
-		if !folderExists(siteDir) {
-
-			// repo
-			siteDir = filepath.Clean(filepath.Join(execPath, "../../webserver/site"))
-			log.Debugf("Looking for site in %v", siteDir)
-			if !folderExists(siteDir) {
-
-				return nil, fmt.Errorf("no HTML template files found. "+
-					"Place the 'site' folder in the executable's directory %q or the working directory, "+
-					"or run dexc from within the client/cmd/dexc source workspace folder.", execPath)
-			}
+	var siteDir string
+	absDir, _ := filepath.Abs("site")
+	for _, dir := range []string{
+		customSiteDir,
+		filepath.Join(execPath, "site"),
+		absDir,
+		filepath.Clean(filepath.Join(execPath, "../../webserver/site")),
```

Comment:
A couple function calls and some slice allocation that otherwise probably wouldn't happen, but a small price to pay to get out of the nesting situation.